packages/contracts-bedrock/test/vendor/eas/EAS.t.sol

```diff
@@ -0,0 +1,2758 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { Test } from "forge-std/Test.sol";
+import { EAS } from "src/vendor/eas/EAS.sol";
+import { SchemaRegistry } from "src/vendor/eas/SchemaRegistry.sol";
+import { Attestation, AttestationRequest, AttestationRequestData, MultiAttestationRequest, RevocationRequest, RevocationRequestData, MultiDelegatedAttestationRequest, MultiDelegatedRevocationRequest, DelegatedAttestationRequest, MultiRevocationRequest, Signature } from "src/vendor/eas/IEAS.sol";
+import { ISchemaResolver } from "src/vendor/eas/resolver/ISchemaResolver.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { console } from "forge-std/console.sol";
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+// Add this helper contract at the top of the file
+contract TestEIP712Helper is EIP712 {
+    constructor() EIP712("EAS", "1.3.0") {}
+
+    function hashTypedDataV4(bytes32 structHash) public view returns (bytes32) {
+        return _hashTypedDataV4(structHash);
+    }
+}
+
+contract MockPayableResolver is ISchemaResolver {
+    function isPayable() external pure override returns (bool) {
+        return true;
+    }
+
+    function attest(
+        Attestation calldata
+    ) external payable override returns (bool) {
+        return true;
+    }
+
+    function multiAttest(
+        Attestation[] calldata,
+        uint256[] calldata
+    ) external payable override returns (bool) {
+        return true;
+    }
+
+    function revoke(
+        Attestation calldata
+    ) external payable override returns (bool) {
+        return true;
+    }
+
+    function multiRevoke(
+        Attestation[] calldata,
+        uint256[] calldata
+    ) external payable override returns (bool) {
+        return true;
+    }
+}
+
+contract EASTest is Test {
+    // =============================================================
+    //                           CONSTANTS
+    // =============================================================
+    uint64 constant NO_EXPIRATION = 0;
+    bytes32 constant ZERO_BYTES32 = bytes32(0);
+    bytes32 private constant ATTEST_TYPEHASH =
+        0xfeb2925a02bae3dae48d424a0437a2b6ac939aa9230ddc55a1a76f065d988076;
+    bytes32 private constant REVOKE_TYPEHASH =
+        0x4e1c85c87bc4c1867b4225cc3fb634a4e0fd8a91feb1ebca195aeaf6611a773b;
+
+    // =============================================================
+    //                        ERROR SELECTORS
+    // =============================================================
+    bytes4 constant InvalidRegistrySelector =
```

Comment:
This is the same as `error InvalidRegistry()` - its better to use the higher level syntax rather than the hash + `bytes4`, the compiler turns it into the `bytes4` under the hood